accumulo-store/src/main/java/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityBloomFilterIterator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gaffer.accumulostore.key.core.impl.byteEntity;
+
+import gaffer.accumulostore.key.core.AbstractCoreKeyBloomFilterIterator;
+import gaffer.accumulostore.utils.ByteArrayEscapeUtils;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+
+
+import java.util.Arrays;
+
+/**
+ * The BloomFilterIterator should filter out elements based on their membership
+ * of the provided bloomFilter. This implementation may not work as desired
+ * depending on your {@link gaffer.accumulostore.key.AccumuloKeyPackage}
+ * implementation.
+ */
+public class ByteEntityBloomFilterIterator extends AbstractCoreKeyBloomFilterIterator {
+
+    private boolean entity;
+    private int pos = -1;
```

Comment:
Because we want to return the position but also do a check on wether what is returned is an entity or edge the check is identical for both cases but you would need to return a complex type if you don't manipulate the variable